src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -1648,9 +1662,21 @@ because xbuild doesn't support framework reference assemblies.
       />
 </Target>
 
+<Target Name="_RunAotForAllRIDs"
+        Condition=" '$(_AndroidUseMarshalMethods)' == 'True' "
```

Comment:
Should this also check that AOT is enabled? We're still using `$(AotAssemblies)` for this. I think there is some overhead when running the `<MSBuild/>` task, because it evaluates the project again (could be like 50ms).

Should we also put this in `Microsoft.Android.Sdk.Aot.targets`, so the target won't exist in "classic" Xamarin.Android?

Reply:
> Should this also check that AOT is enabled? We're still using `$(AotAssemblies)` for this. I think there is some overhead when running the `<MSBuild/>` task, because it evaluates the project again (could be like 50ms).
> 

The `_AndroidAot` checks `$(AotAssemblies)`, I think it's enough - it will just return no items if AOT is disabled.

> Should we also put this in `Microsoft.Android.Sdk.Aot.targets`, so the target won't exist in "classic" Xamarin.Android?

We could, but then we need to make reference to this target from `_GeneratePackageManagerJavaDependsOn` optional, not sure if it's worth it?

Reply:
Does this target run on a Debug build? How long does it take?

Reply:
Nope, it doesn't. `$(_AndroidUseMarshalMethods)` [guarantees it](https://github.com/xamarin/xamarin-android/pull/7351/files#diff-e2c60a31385c6fbadf9b9f439d3681ee357077bf3e66f181a0ddb02b9d98adf1R338).